cpp/modmesh/buffer/pymod/wrap_SimpleArray.cpp

```diff
@@ -87,10 +88,37 @@ class MODMESH_PYTHON_WRAPPER_VISIBILITY WrapSimpleArray
                         const bool is_c_contiguous = (arr_in.flags() & py::array::c_style) == py::array::c_style;
                         const bool is_f_contiguous = (arr_in.flags() & py::array::f_style) == py::array::f_style;
 
-                        std::shared_ptr<ConcreteBuffer> const buffer = ConcreteBuffer::construct(
-                            arr_in.nbytes(),
-                            arr_in.mutable_data(),
-                            std::make_unique<ConcreteBufferNdarrayRemover>(arr_in));
+                        py::array owner = arr_in;
+                        while (true)
+                        {
+                            const py::object b = owner.attr("base");
+                            if (b.is_none() || !py::isinstance<py::array>(b))
+                            {
+                                break;
+                            }
+                            auto next = b.cast<py::array>();
+                            if (next.ptr() == owner.ptr())
+                            {
+                                break;
+                            }
+                            owner = next;
+                        }
+
+                        char * base_ptr = static_cast<char *>(owner.mutable_data());
+                        char * view_ptr = static_cast<char *>(arr_in.mutable_data());
+                        const ptrdiff_t offset_bytes = view_ptr - base_ptr;
```

Comment:
It seems that you piggybacked a fix to issue #527 in this PR without saying it explicitly.  Thanks for the fix, but it is not good to mix two things in one PR.  Please separate this fix (to issue #527, along with the related unit test code) in a different PR.